SonarQube.Bootstrapper/ArgumentProcessor.cs

```diff
@@ -67,19 +103,58 @@ public static bool TryProcessArgs(string[] commandLineArgs, ILogger logger, out
                 Debug.Assert(globalFileProperties != null);
                 IAnalysisPropertyProvider properties = new AggregatePropertiesProvider(cmdLineProperties, globalFileProperties);
 
+                parsedOk = TryCreatePhaseSpecificSettings(phase, properties, logger, out settings);
+            }
+
+            return settings != null;
+        }
+
+        private static bool TryGetPhase(int originalArgCount, IEnumerable<ArgumentInstance> arguments, ILogger logger, out AnalysisPhase phase)
+        {
+            // The command line parser will already have checked for duplicates
+            ArgumentInstance argumentInstance;
+            bool hasBeginVerb = ArgumentInstance.TryGetArgument(BeginId, arguments, out argumentInstance);
+            bool hasEndVerb = ArgumentInstance.TryGetArgument(EndId, arguments, out argumentInstance);
+
+            if (hasBeginVerb && hasEndVerb) // both
+            {
+                phase = AnalysisPhase.Unspecified;
+                logger.LogError(Resources.ERROR_CmdLine_BothBeginAndEndSupplied);
+            }
+            else if (!hasBeginVerb && !hasEndVerb) // neither
+            {
+                // Backwards compatibility - decide the phase based on the number of arguments passed
+                phase = originalArgCount == 0 ? AnalysisPhase.PostProcessing : AnalysisPhase.PreProcessing;
+                logger.LogWarning(Resources.WARN_CmdLine_v09_Compat);
```

Comment:
Backwards compat: allow the verbs to be omitted, but issue a warning.
There is a specific unit tests that tests that the v0.9 syntax still works (although the user will need to supply a default properties settings file next to the bootstrapper).